A88-xds-data-error-handling.md

```diff
@@ -0,0 +1,529 @@
+[gRFC A71](A71-xds-fallback.md).  However, because we do not want to
+trigger fallback due to an individual resource not existing if the control
+plane is actually responding, we do not currently trigger fallback when
+the resource timer fires if we are still connected to the control plane.
+This has led to concerns about failure modes where control plane slowness
+may cause clients to act as if resources do not exist rather than using a
+fallback server.  Ideally, it seems like that we would like to use a timer
+to consider the xDS server connection to have failed if we haven't gotten
+back a response in a certain amount of time, but we cannot do that while
+we are depending on the timer to detect individual resources not existing.
+
+As future work, we should consider whether there are ways to safely
+trigger xDS fallback if the primary server is responding too slowly.
+There are a number of challenges here:
+
+- Given the number of issues we've seen where the control plane has been
+  slow to respond, it's not clear to me that we would want all of those
+  cases to trigger fallback, since we'd likely hit fallback way too often.
+  I suspect that we would first need the control plane to monitor and provide
+  an SLO for response time upon initial subscription to a resource.
+
+- The current timer semantics are that we start a separate timer for each
+  individual resource when it is initially subscribed to.  This ensures
+  that any control plane slowness that affects the client's functionality
+  at any time during the stream will cause us to see a problem.
+  However, given that our criteria for switching back to the primary
+  server is when we get a single resource back from the server, there
+  is an incompatibility here: if the control plane is sending back at
+  least one resource but is timing out for others, then we will have
+  simultaneously met the criteria for using fallback and the criteria
+  for switching back to the primary server.
+
+- Given the above, it might seem beneficial to consider changing the timer
+  semantics to just check for a timeout on any xDS response, rather
+  than having a separate timer per resource.  Unfortunately, this won't
+  actually work, because the xDS protocol does not require any minimum
+  interval between responses.  Once the server has sent all of the
+  resources to the client, if those resources never change, then the
+  server never needs to send another response, and we would not want to
+  trigger fallback in this case.
+
+- As a middle ground, we could consider only having a timeout on the very
+  first response on the ADS stream to trigger fallback, or even the
+  very first response for each resource type.  However, this would
+  still leave us blind to cases where the control plane is slow when new
+  subscriptions are added later in the stream's lifetime, which we have
+  seen in the wild.
+
+- Even if we could identify the right timeout criteria to trigger fallback,
```

Comment:
Or on second thought, maybe this RFC isn't the right place for all of the details of these complications.  We could leave them in our discussion notes & in the PR description instead?